can/broadcastmanager.py

```diff
@@ -120,23 +135,25 @@ def stop(self):
     def start(self):
         self.stopped = False
         if self.thread is None or not self.thread.is_alive():
-            name = "Cyclic send task for 0x%X" % (self.message.arbitration_id)
+            name = "Cyclic send task for 0x%X" % (self.messages[0].arbitration_id)
             self.thread = threading.Thread(target=self._run, name=name)
             self.thread.daemon = True
             self.thread.start()
 
     def _run(self):
+        msg_index = 0
         while not self.stopped:
             # Prevent calling bus.send from multiple threads
-            with self.lock:
+            with self.send_lock:
                 started = time.time()
                 try:
-                    self.bus.send(self.message)
+                    self.bus.send(self.messages[msg_index])
```

Comment:
There is a chance here that the `msg_index` is no longer valid if `self.messages` is modified.

Reply:
The Linux SocketCAN BCM APIs require that the new array size must be equal to the given array size. Is this a restriction that we also want to impose in the generic implementation?

Reply:
That seems a reasonable restriction